### PR TITLE
APIS-4691

### PR DESCRIPTION
**Pull request by AndySpaven**

### CODE REVIEW

File: app/uk/gov/hmrc/apipublisher/controllers/PublisherController.scala [hunk 1]
```diff
@@ -55,17 +56,45 @@ class PublisherController @Inject()(publisherService: PublisherService,
...
+
+    def publish(apiAndScopes: ApiAndScopes): Future[Result] = {
+      publisherService.publishAPIDefinitionAndScopes(serviceLocation, apiAndScopes).map {
+        case Some(true) =>
+          Logger.info(s"Successfully published API Definition and Scopes for ${serviceLocation.serviceName}")
+          NoContent
+        case Some(false) =>
+          Logger.info(s"Publication awaiting approval for ${serviceLocation.serviceName}")
+          Accepted
+        case None =>
```
Comment (lburgos) [inline app/uk/gov/hmrc/apipublisher/controllers/PublisherController.scala:84]: I think this branch is unreachable now. It used to be none if the service didn't have an api definition but we're not calling that endpoint from the publisher service anymore (it used to be in line 70 of the publisher service in the old code)

File: app/uk/gov/hmrc/apipublisher/controllers/PublisherController.scala [hunk 2]
```diff
@@ -55,17 +56,45 @@ class PublisherController @Inject()(publisherService: PublisherService,
...
+    type Over[A] = EitherT[Future,Result, A]
+
+    def getDefinition: Over[ApiAndScopes] = {
+      EitherT(definitionService.getDefinition(serviceLocation).map(_.toRight(BadRequest)))
+    }
+
+    def validate(apiAndScopes: ApiAndScopes): Over[Unit] = {
+      EitherT.fromOptionF(
+        OptionT(publisherService.validateAPIDefinitionAndScopes(apiAndScopes)).map { e =>
+          Console.println("***** "+e)
```
Comment (lburgos) [inline app/uk/gov/hmrc/apipublisher/controllers/PublisherController.scala]: Should we use the logger?

File: test/uk/gov/hmrc/apipublisher/controllers/PublisherControllerSpec.scala [hunk 3]
```diff
@@ -43,66 +43,99 @@ import org.mockito.Matchers.{eq => eqTo, any}
 
 class PublisherControllerSpec extends UnitSpec with MockitoSugar with GuiceOneAppPerSuite with StubControllerComponentsFactory{
 
-  private val serviceLocation = ServiceLocation("TestService", "http://test.example.com")
+  val serviceLocation = ServiceLocation("test", "http://example.com", Some(Map("third-party-api" -> "true")))
+  // private val serviceLocation = ServiceLocation("TestService", "http://test.example.com")
```
Comment (lburgos) [inline test/uk/gov/hmrc/apipublisher/controllers/PublisherControllerSpec.scala:47]: Looks like we can remove this and a couple more down below

File: app/uk/gov/hmrc/apipublisher/controllers/PublisherController.scala (hunk 2 above)
Comment (AndySpaven) [reply app/uk/gov/hmrc/apipublisher/controllers/PublisherController.scala]: I've removed it altogether - was only in use for a debug.

File: app/uk/gov/hmrc/apipublisher/controllers/PublisherController.scala (hunk 1 above)
Comment (lburgos) [reply app/uk/gov/hmrc/apipublisher/controllers/PublisherController.scala:84]: If that's true, we may also want to turn the option of boolean into just a boolean

Comment (AndySpaven) [reply app/uk/gov/hmrc/apipublisher/controllers/PublisherController.scala:84]: Agreed.  Will trim the types.

**Pull request merged**